convert_checkpoints.py

```diff
@@ -462,6 +471,99 @@ def _get_gemma_state_dict(input_ckpt_dir):
   return state_dict, model_config
 
 
+def _get_mixtral_state_dict(input_ckpt_dir):
+  ckpt_files = list(input_ckpt_dir.glob("*.pt"))
+  assert len(ckpt_files) == 8, "only expect 8 ckpt file for Mistral model."
+
+  start = time.perf_counter()
+  state_dict = {}
+  for file in sorted(ckpt_files):
+    ckpt = torch.load(
+        str(file), map_location="cpu", mmap=True, weights_only=True
+    )
+    state_dict.update(ckpt)
+  end = time.perf_counter()
+  print(f"Loading checkpoints takes {end - start} seconds")
+
+  for k, v in state_dict.items():
+    if "layers" in k and "layers.0" not in k:
+      continue
+    print(f"The loaded key: {k} and value: {v.shape} {v.dtype}")
+
+  config = json.loads((input_ckpt_dir / "config.json").read_text())
+  print(f"Loaded config: {config}")
+  weight_map = {
+      "tok_embeddings.weight": "tok_embeddings.weight",
+      "layers.{}.attention.wq.weight": "layers.{}.attention.wq.weight",
+      "layers.{}.attention.wk.weight": "layers.{}.attention.wk.weight",
+      "layers.{}.attention.wv.weight": "layers.{}.attention.wv.weight",
+      "layers.{}.attention.wo.weight": "layers.{}.attention.wo.weight",
+      "layers.{}.block_sparse_moe.w1": "layers.{}.block_sparse_moe.cond_ffn.w1",
```

Comment:
Looks like only these weight name are difference, can we only store the the different name in the map?

Reply:
Good point, removed